crypto/keyring/keyring.go

```diff
@@ -320,47 +319,36 @@ func (ks keystore) ImportPubKey(uid string, armor string) error {
 		return err
 	}
 
-	_, err = ks.writeOfflineKey(uid, pubKey, hd.PubKeyType(algo))
+	_, err = ks.writeOfflineKey(uid, pubKey)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
+
 // ImportInfo implements Importer.MigrateInfo.
-func (ks keystore) ImportInfo(oldInfo Info) error {
+// TODO do i need it or not
```

Comment:
I would say we don't need. I think it's for 0.39->0.40 keyring migration (which you wrote the guide here: https://docs.cosmos.network/v0.42/migrations/keyring.html)